io/io/inc/TMemFile.h

```diff
@@ -19,8 +19,14 @@
 class TMemFile : public TFile {
 public:
    using ExternalDataPtr_t = std::shared_ptr<const std::vector<char>>;
+   /// A read-only memory range which we do not control.
+   struct ExternalDataRange_t {
+      const char *fStart;
+      const size_t fSize;
+      ExternalDataRange_t(const char * start, const size_t size) : fStart(start), fSize(size) {}
+   };
 
-private:
+protected:
```

Comment:
Why does it need to become protected?

Reply:
See the test case, it relies on checking if the internal  pointer is same as the one in the blob. To do that we need to access that data and we cannot do it if it is private.

Reply:
You could use `TDataMember`.

Reply:
That would mean going through several layers of indirection and relying on reflection. That seems to make the test significantly more complex to debug. I'd prefer not taking that route.